### PR TITLE
[freebox] Adding Fiber optic and support of Delta bridge

### PR DESCRIPTION
**Pull request by clinique**

### CODE REVIEW

File: bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/FreeboxApiManager.java [hunk 1]
```diff
@@ -201,6 +202,10 @@ public String getxDslStatus() throws FreeboxException {
         return executeGetUrl("connection/xdsl/", FreeboxXdslStatusResponse.class).getStatus();
     }
 
+    public String getFtthStatus() throws FreeboxException {
+        return executeGetUrl("connection/ftth/", FreeboxFtthStatusResponse.class).getSfp_present().toString();
```
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/FreeboxApiManager.java]: This was more intended to introduce a quick running solution to solve F.O. issue. I plan a wider review later on this binding to introduce other field and switch to UoM

File: bundles/org.openhab.binding.freebox/README.md [hunk 2]
```diff
@@ -129,6 +129,7 @@ The following channels are supported:
 | server        | sambafileshare_status    | Switch    | RW          | Indicates whether Window File Sharing  is enabled                               |
 | server        | sambaprintershare_status | Switch    | RW          | Indicates whether Window Printer Sharing  is enabled                            |
 | server        | xdsl_status              | String    | R           | Status of the xDSL line                                                         |
+| server        | ftth_status              | String    | R           | Status of the Ftth line                                                         |
```
Comment (lolodomo) [inline bundles/org.openhab.binding.freebox/README.md]: Please update from String to Switch.

File: bundles/org.openhab.binding.freebox/README.md [hunk 3]
```diff
@@ -189,7 +190,8 @@ Bridge freebox:server:fb "Freebox Revolution" [ fqdn="abcdefgh.fbxos.fr", appTok
 ### Items
 
 ```java
-String Freebox_xdsl_status "Freebox state [%s]" {channel="freebox:server:fb:xdsl_status"}
+String Freebox_xdsl_status "Freebox ADSL state [%s]" {channel="freebox:server:fb:xdsl_status"}
+String Freebox_ftth_status "Freebox Fiber Optic state [%s]" {channel="freebox:server:fb:ftth_status"}
```
Comment (lolodomo) [inline bundles/org.openhab.binding.freebox/README.md]: Please update from String to Switch and suppress `[%s]`

File: bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/model/FreeboxFtthStatus.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    public boolean getSfpAlimOk() {
+        return sfp_alim_ok;
+    }
+
+    public boolean getSfpHasPowerReport() {
+        return sfp_has_power_report;
+    }
+
+    public Boolean getSfpHasSignal() {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/model/FreeboxFtthStatus.java]: Please update the return type from Boolean to boolean

File: bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/model/FreeboxFtthStatus.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        return sfp_has_signal;
+    }
+
+    public Boolean getLink() {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/model/FreeboxFtthStatus.java]: Please update the return type from Boolean to boolean

**Comment by lolodomo** [comment]
Anyone you merge this PR? 
I already reviewed it.

File: bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/model/FreeboxFtthStatusResponse.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import org.openhab.binding.freebox.internal.api.FreeboxException;
+
+/**
+ * The {@link FreeboxFtthStatusResponse} is the Java class used to map the
+ * response of the connection Ftth status API
+ * https://dev.freebox.fr/sdk/os/connection/#
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+public class FreeboxFtthStatusResponse extends FreeboxResponse<FreeboxFtthStatus> {
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/model/FreeboxFtthStatusResponse.java]: What about nullness annotations in the new classes?

**Comment by lolodomo** [comment]
Good point, the binding description and the bridge type label have to be updated.

You could even add a short sentence in the first chapter of the documentation to mention that the binding is working whatever the connection (xDSL and fiber optic).

File: bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/model/FreeboxFtthStatusResponse.java (hunk 6 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/model/FreeboxFtthStatusResponse.java]: This is a very old binding without nullness annotations.
And the class you point  is an object for JSON mapping, The nonnull annotation is not very appropriate for such case.

Comment (cweitkamp) [reply bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/model/FreeboxFtthStatusResponse.java]: Agreed.

**Pull request merged**